pennylane/devices/qubit/simulate.py

```diff
@@ -335,6 +362,361 @@ def simulate_partial(k):
+                return i, op
+        return len(circuit.operations) + 1, None  # pragma: no cover
+
+    i, op = find_next_mcm(circuit)
+    # run circuit until next MidMeasureMP and sample
+    circuit_base = qml.tape.QuantumScript(
+        circuit.operations[0:i],
+        [qml.sample(wires=op.wires) if op.obs is None else qml.sample(op=op.obs)],
+        shots=circuit.shots,
+        trainable_params=circuit.trainable_params,
+    )
+    # circuit beyond next MidMeasureMP with VarianceMP <==> SampleMP
+    new_measurements = []
+    for m in circuit.measurements:
+        if not m.mv:
+            if isinstance(m, VarianceMP):
+                new_measurements.append(SampleMP(obs=m.obs))
+            else:
+                new_measurements.append(m)
+    circuit_next = qml.tape.QuantumScript(
+        circuit.operations[i + 1 :],
+        new_measurements,
+        shots=circuit.shots,
+        trainable_params=circuit.trainable_params,
+    )
+    return circuit_base, circuit_next, op
+
+
+def measurement_with_no_shots(measurement):
+    """Returns a NaN scalar or array of the correct size when executing an all-invalid-shot circuit."""
+    if isinstance(measurement, ProbabilityMP):
+        if measurement.obs:
+            return np.nan * np.ones(measurement.obs.eigvals().size)
+        return np.nan * np.ones(2 ** len(measurement.wires))
+    return np.nan
+
+
+def combine_measurements(circuit, measurements, mcm_samples):
+    """Returns combined measurement values of various types."""
+    empty_mcm_samples = len(next(iter(mcm_samples.values()))) == 0
+    if empty_mcm_samples and any(len(m) != 0 for m in mcm_samples.values()):  # pragma: no cover
+        raise ValueError("mcm_samples have inconsistent shapes.")
+    # loop over measurements
+    final_measurements = []
+    for circ_meas in circuit.measurements:
+        if circ_meas.mv and empty_mcm_samples:  # pragma: no cover
+            comb_meas = measurement_with_no_shots(circ_meas)
+        elif circ_meas.mv:
+            mcm_samples = dict((k, v.reshape((-1, 1))) for k, v in mcm_samples.items())
+            is_valid = qml.math.ones(list(mcm_samples.values())[0].shape[0], dtype=bool)
+            comb_meas = dyn_gather_mcm(circ_meas, mcm_samples, is_valid)
+        elif not measurements or not measurements[0]:  # pragma: no cover
+            if len(measurements) > 0:
+                _ = measurements.pop(0)
+            comb_meas = measurement_with_no_shots(circ_meas)
+        else:
+            comb_meas = combine_measurements_core(circ_meas, measurements.pop(0))
+        if isinstance(circ_meas, SampleMP):
+            comb_meas = qml.math.squeeze(comb_meas)
+        final_measurements.append(comb_meas)
+    # special treatment of var
+    for i, (c, m) in enumerate(zip(circuit.measurements, final_measurements)):
+        if not c.mv and isinstance(circuit.measurements[i], VarianceMP):
+            final_measurements[i] = qml.math.var(m)
```

Comment:
Because the variance requires global information (the mean) which cannot be computed at the leafs. This is the same for `one-shot` where the variance of a single shot is always 0, but the global variance isn't.

Reply:
Thanks, that makes sense. Resolving.